test/acceptance/failures_test.js

```diff
@@ -427,7 +428,25 @@ describe('Endpoint Failures', function() {
         title: "title1",
         color: "#000000",
         logoPath: "https://example.com/logo.png",
-        removeOverlay: false
+        removeOverlay: false,
+        customDomain: 'hey.abc.com'
+      };
+      const options = { method: 'PUT', url: '/admin/settings', headers, payload };
+ 
+      const res = await server.inject(options);
+      expect(res.statusCode).to.equal(400);
+      expect(res.result).to.deep.equal({ 
+        error: 'Bad Request', 
+        message: 'Invalid request payload input', 
+        statusCode: 400 
+      });
+    });
+    it('returns a 400 with an invalid payload - missing multiple properties', async function() {
```

Comment:
Can we add a test covering the integration with Management API custom domain validation?

Reply:
I added a test mocking a failure of a match. Actually building integration with Management API might take more time.

Reply:
I think you should be able to nock out that interaction easy enough. Not necessarily looking for a full system test

Reply:
Ah right good call

Reply:
I guess the request library doesn't work with nock because it's not made using Node.js's built-in https/http modules

Reply:
Was looking into swapping with axios but the wrapper of request (apiCall) is used in the rule/action also. What do you think of a follow up ticket removing `request` for `axios`?

Reply:
Looking in api2. I guess `request` package should work with `nock`. Going to figure out whats going on

Reply:
Got it working, pushing up now